### PR TITLE
Bp 352 caching

### PR DESCRIPTION
**Pull request by MaximilianV**
Implement a basic cache for api request.

### CODE REVIEW

File: src/Utils/ConnectionCache.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	}
+
+	static invalidate(url) {
+		const currentCache = ConnectionCache.getStorageCache();
+		const newCache = currentCache.delete(url);
+		ConnectionCache.writeStorageCache(newCache);
+	}
+
+	static invalidateFromNotification(artifactType, artifactId) {
+		let currentCache = ConnectionCache.getStorageCache();
```
Comment (hpibptlab) [inline src/Utils/ConnectionCache.js]: MINOR Make "currentCache" "const". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS3353)

File: src/Utils/ConnectionCache.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+class ConnectionCache {
+
+	static add(cache) {
+		const url = cache.url;
+		cache.json().then(data => ConnectionCache.addResolvedRequest(url, data));
+	}
+
+	static addResolvedRequest(url, data) {
+		let currentCache = ConnectionCache.getStorageCache();
```
Comment (hpibptlab) [inline src/Utils/ConnectionCache.js]: MINOR Make "currentCache" "const". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS3353)

File: src/Utils/ConnectionComponent.js [hunk 3]
```diff
@@ -32,20 +33,55 @@ class ConnectionComponent extends Component {
 		return null;
 	}
 
+	static handleMockRequest(request) {
+		const mockData = RestRoutesManager.executeMockFunction(request);
+		return ConnectionComponent.buildResponse(mockData);
+	}
+
+	static cacheResponse(response) {
+		if (!response.url ||
```
Comment (hpibptlab) [inline src/Utils/ConnectionComponent.js]: MAJOR Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1067)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Review by friedow — APPROVED** [review]

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Pull request merged**